templates/decays.cpp.in

```diff
@@ -0,0 +1,294 @@
+}
+
+// eq. 2.6 of hep-ph/0503173
+double calc_DeltaH(double b)
+{
+   return calc_A(b)/b + 1./(16*Power3(b)) * (
+      3 + 34*b*b - 13*Power4(b)) * std::log((1+b)/(1-b)) + 3./(8.*b*b)*(7*b*b-1);
+}
+
+// eq. 2.6 of hep-ph/0503173
+double calc_DeltaAH(double b)
+{
+   return calc_A(b)/b + 1./(16*b) * (19 + 2*b*b + 3*Power4(b))
+      * std::log((1+b)/(1-b)) + 3./8.*(7-b*b);
+}
+
+// 2 and higher order
+double calc_deltaqq(double alpha_s_red, double Nf)
+{
+   return
+      // order alphas are taken into account with mass sependence somewhere alse
+      0. * 17./3. * alpha_s_red
+      + (35.94 - 1.36 * Nf) * Sqr(alpha_s_red)
+      + (164.14 - 25.77 * Nf + 0.259 * Sqr(Nf)) * Power3(alpha_s_red)
+      + (39.34 - 220.9 * Nf + 9.685 * Sqr(Nf) - 0.0205 * Power3(Nf)) * Power4(alpha_s_red);
+}
+
+// eq. 2.31 of hep-ph/0503172
+double RT(double x) {
+   BOOST_ASSERT_MSG(x<=1, "This function");
+   if (is_zero(x-1)) {
+      return 0.;
+   } else if (1-x < 1e-6) {
+      return 39*(-1 + x) + (75*Sqr(-1 + x))/2. - 6*Power3(-1 + x);
+   } else {
+    return 3.*(1. - 8.*x + 20.*x*x)/std::sqrt(4.*x-1.) * std::acos((3.*x-1.)/(2.*std::pow(x, 3./2.)))
+    - (1.-x)/(2.*x) * (2. - 13.*x + 47.*x*x)
+    - 3./2. * (1. - 6.*x + 4.*x*x) * std::log(x);
+   }
+}
+
+} // anonymous namespace
+
+@partialWidthSpecializationDefs@
+
+template <>
+double CLASSNAME::get_partial_width<H, A, A>(
+      const context_base& context,
+      const typename field_indices<H>::type& in_idx,
+      const typename field_indices<A>::type& out1_idx,
+      const typename field_indices<A>::type& out2_idx) const
+{
+   const auto amp = calculate_amplitude<H, A, A>(context, in_idx, out1_idx, out2_idx);
+   const double mH = context.physical_mass<H>(in_idx);
+   constexpr double ps {1./(8.*Pi)};
+   constexpr double ps_symmetry {1./2.};
+   const double color_gen_sqr = squared_color_generator<H, A, A>();
+   const double flux = 0.5/mH;
+   const auto res = flux * color_gen_sqr * ps * ps_symmetry * amp.square();
+
+   return res;
+}
+
+template <>
+double CLASSNAME::get_partial_width<H, A, Z>(
+      const context_base& context,
+      const typename field_indices<H>::type& in_idx,
+      const typename field_indices<A>::type& out1_idx,
+      const typename field_indices<Z>::type& out2_idx) const
+{
+   const double mH = context.physical_mass<H>(in_idx);
+   const double mA = context.physical_mass<A>(out1_idx);
+   const double mZ = context.physical_mass<Z>(out2_idx);
+
+   const double ps {1./(8.*Pi) * sqrt(KallenLambda(1., Sqr(mA/mH), Sqr(mZ/mH)))};
+   constexpr double ps_symmetry {1.};
+   const double color_gen_sqr = squared_color_generator<H, A, Z>();
+   const double flux = 0.5/mH;
+
+   const auto amp = calculate_amplitude<H, A, Z>(context, in_idx, out1_idx, out2_idx);
+
+   const auto res = flux * color_gen_sqr * ps * ps_symmetry * amp.square();
+
+   return res;
+}
+
+template <>
+double CLASSNAME::get_partial_width<H, G, G>(
```

Comment:
This function seems to be identical to the one defined in `src/H_SM_decays/decay_AH_to_GG.cpp`. If this is supposed to be correct, then this function should be removed and one should use the one defined in `src/H_SM_decays/decay_AH_to_GG.cpp`.

Reply:
This is H->GG. `src/H_SM_decays/decay_AH_to_GG.cpp` contains AH->GG.

Reply:
Ok, but it seems to me the content of the two functions is identical. That's why I'm confused. Isn't there a difference between these two processes?

Reply:
Originally pseudoscalar decays were not part of the project. It's possible that for now the body of the AH->GG function is just a copy of the H->GG one.

Reply:
Hmm, does this mean, that the body of the function is just (or most probably) wrong? If so, please remove the wrong part, such that the function returns a correct leading order result.